components/tidb_query/src/codec/table.rs

```diff
@@ -76,15 +75,14 @@ pub fn check_table_ranges(ranges: &[KeyRange]) -> Result<()> {
 
 /// Decodes table ID from the key.
 pub fn decode_table_id(key: &[u8]) -> Result<i64> {
```

Comment:
here is why I introduced a local mutable reference `buf` instead of using the argments to `read_bytes`

when  read table prefix fails, the original  implementation will report error  contains what key is
```
return Err(invalid_type!("record key expected, but got {}",hex::encode_upper(key)));
```
if we use the argument `key` to read_bytes,  it will lose the bytes we had read when return error.
or 
1 . we concat bytes  back ` hex::encode_upper( read_prefix + key)`
```
return Err(invalid_type!("record key expected, but got {}", hex::encode_upper( prefix + key)));
    ....
return Err(invalid_type!("record key expected, but got {}", hex::encode_upper(prefix  + table_id +  sep_prefix + encoded)));
```

2.  we  only report error part, eg: 
```
return Err(invalid_type!("invalid table prefix, {}", hex::encode_upper(prefix))
```

Reply:
I didn't considered this part. You are doing right!